platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/internal/DefaultJavaPluginExtension.java

```diff
@@ -258,9 +254,12 @@ public JavaToolchainSpec toolchain(Action<? super JavaToolchainSpec> action) {
     public void consistentResolution(Action<? super JavaResolutionConsistency> action) {
         maybeEmitMissingJavaComponentDeprecation("consistentResolution(Action)");
 
-        final ConfigurationContainer configurations = project.getConfigurations();
-        final SourceSetContainer sourceSets = getSourceSets();
-        action.execute(project.getObjects().newInstance(DefaultJavaPluginExtension.DefaultJavaResolutionConsistency.class, project, sourceSets, configurations));
+        project.getComponents().configureEach(component -> {
```

Comment:
Why not use `withType`?

Reply:
Maybe the same argument as above applies here?  Shouldn't this only affect the `java` component?  I'm going to change this to just affect the `java` component.

Reply:
Discussed.  Going to configure each `JvmSoftwareComponentInternal` and not privilege the `java` component.